modules/lib/lib-project/src/test/resources/test/ModifyProjectPermissionsHandlerTest.js

```diff
@@ -0,0 +1,156 @@
+var assert = require('/lib/xp/testing.js');
+var project = require('/lib/xp/project.js');
+
+var addPermissionsExpected = {
+    permissions: {
+        owner: [
+            "user:system:owner2",
+            "user:system:owner1"
+        ],
+        viewer: [
+            "user:system:viewer1"
+        ]
+    }
+};
+
+exports.addPermissions = function () {
+    createProject();
+
+    var result = project.addPermissions({
+        name: 'myproject',
+        permissions: {
+            owner: [
+                "user:system:owner2",
+                "user:system:owner1"
+            ],
+            viewer: [
+                "user:system:viewer1"
+            ]
+        }
+    });
+
+
+    assert.assertJsonEquals(addPermissionsExpected, result);
+};
+
+exports.addPermissionsNull = function () {
+    createProject();
+
+    var result = project.addPermissions({
+        name: 'myproject',
+        permissions: null
+    });
+
+
+    assert.assertJsonEquals({permissions: {}}, result);
+
+    result = project.addPermissions({
+        name: 'myproject',
+        permissions: {owner: ["user:system:owner"]}
+    });
+
+
+    assert.assertJsonEquals({permissions: {owner: ["user:system:owner"]}}, result);
+
+    result = project.addPermissions({
+        name: 'myproject',
+        permissions: {owner: null, author: null}
+    });
+
+    assert.assertJsonEquals({permissions: {owner: ["user:system:owner"]}}, result);
+};
+
+var removePermissionsExpected = {
+    permissions: {
+        owner: [
+            "user:system:owner3"
+        ]
+    }
+};
+
+exports.removePermissions = function () {
+    createProject();
+
+    addPermissions(
+        'myproject',
+        {
+            owner: [
+                "user:system:owner3",
+                "user:system:owner2",
+                "user:system:owner1"
+            ],
+            viewer: [
+                "user:system:viewer1"
+            ]
+
+        });
+
+    var result = removePermissions('myproject', {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['removePermissions' was used before it was defined.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5502328)